File: app/llm/train.py
import os
import pathlib
import pickle
import shutil

import dotenv
from langchain import hub
from langchain_chroma import Chroma
from langchain_community.document_loaders import TextLoader
from langchain_openai import OpenAIEmbeddings
from langchain_text_splitters import MarkdownTextSplitter

dotenv.load_dotenv()


def main():
    if os.path.isdir("./data/store/chromadb/"):
        shutil.rmtree("./data/store/chromadb/")
    pathlib.Path("./data/store").mkdir(parents=True, exist_ok=True)
    pathlib.Path("./data/objects").mkdir(parents=True, exist_ok=True)
    loader = TextLoader("./Resume.md")
    docs = loader.load()

    md_splitter = MarkdownTextSplitter(chunk_size=900)
    splits = md_splitter.split_documents(docs)
    _ = Chroma.from_documents(
        documents=splits,
        embedding=OpenAIEmbeddings(),
        persist_directory="./data/store/chromadb",
    )

    prompt = hub.pull("rlm/rag-prompt")
    with open("./data/objects/prompt.pkl", "wb") as f:
        pickle.dump(prompt, f)


if __name__ == "__main__":
    main()

File: app/llm/train.js
import { existsSync, mkdir, rm } from "node:fs";
import { TextLoader } from "langchain/document_loaders/fs/text";
import { MarkdownTextSplitter } from "langchain/text_splitter";
import { Chroma } from "@langchain/community/vectorstores/chroma";
import { OpenAIEmbeddings } from "@langchain/openai";

const CHROMA_PATH = "./data/store/chromadb";
const RESUME_PATH = "./Resume.md";

async function main() {
  if (existsSync(CHROMA_PATH)) {
    rm(CHROMA_PATH, { recursive: true });
  }
  mkdir(CHROMA_PATH, { recursive: true }, (error) => {
    throw error;
  });

  const loader = new TextLoader(RESUME_PATH);
  const docs = await loader.load();

  const md_splitter = new MarkdownTextSplitter({ chunkSize: 900 });
  const splits = await md_splitter.splitDocuments(docs);

  const embeddings = new OpenAIEmbeddings();
  const chromadoc = await Chroma.fromDocuments(splits, embeddings);
}
